### PR TITLE
HDDS-4497. Recon File Size Count task throws SQL Exception.

### PR DESCRIPTION
**Pull request by avijayanhwx**
## What changes were proposed in this pull request?
Fix JOOQ truncate call usage in Recon.

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-4497

## How was this patch tested?
Manually tested. Added unit tests.

### CODE REVIEW

File: hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/FileSizeCountTask.java [hunk 1]
```diff
@@ -96,7 +96,9 @@ private static int nextClosestPowerIndexOfTwo(long dataSize) {
       return new ImmutablePair<>(getTaskName(), false);
     }
     // Truncate table before inserting new rows
-    dslContext.truncate(FILE_COUNT_BY_SIZE);
+    int execute = dslContext.delete(FILE_COUNT_BY_SIZE).execute();
```
Comment (swagle) [inline hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/FileSizeCountTask.java:99]: Why does truncate not work?

Comment (avijayanhwx) [reply hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/FileSizeCountTask.java:99]: From Jooq doc.

`  * Most dialects implement the <code>TRUNCATE</code> statement. If it is not
     * supported, it is emulated using an equivalent <code>DELETE</code>
     * statement. This is particularly true for these dialects:
     * <ul>
     * <li> {@link SQLDialect#FIREBIRD}</li>
     * <li> {@link SQLDialect#INGRES}</li>
     * <li> {@link SQLDialect#SQLITE}</li>
     * </ul>`

**Review by swagle — APPROVED** [review]
+1

**Review by vivekratnavel — APPROVED** [review]
+1 LGTM

**Comment by avijayanhwx** [comment]
Thanks for the reviews @swagle & @vivekratnavel.

**Pull request merged**